src/golang/lib/repos/notification.go

```diff
@@ -0,0 +1,39 @@
+package repos
+
+import (
+	"context"
+
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/models"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/google/uuid"
+)
+
+// Notification defines all of the database operations that can be performed for a Notification.
+type Notification interface {
+	notificationReader
+	notificationWriter
+}
+
+type notificationReader interface {
+	// GetByReceiver returns the Workflow with ID.
+	GetByReceiver(ctx context.Context, receiverID uuid.UUID, status shared.Status, DB database.Database) ([]models.Notification, error)
+
+	// ValidateUser returns whether the Notification belongs to userID.
+	ValidateUser(ctx context.Context, notificationID uuid.UUID, userID uuid.UUID, DB database.Database) (bool, error)
```

Comment:
could you clarify what this means in the comment. Are we checking that the `userID` is the receiver of the nofication?